packages/server/src/shared/createProxy/index.ts

```diff
@@ -4,6 +4,20 @@ interface ProxyCallbackOptions {
 }
 type ProxyCallback = (opts: ProxyCallbackOptions) => unknown;
 
+export type ProxyPromiseGuard = {
+  /**
+   * We prevent our procedures from accidentally being treated as a PromiseLike
+   * by always ensuring that "then" resolves to `undefined`.
+   *
+   * This prevents errors where a proxy client is used in a promise callback or
+   * in a `Promise.resolve(proxy)`.
+   *
+   * This does mean that `then` is a reserved word and cannot be used to
+   * name any of your procedures.
+   */
+  then: undefined;
```

Comment:
Also add `catch` and `finally` in here?

Reply:
I don't believe it's necessary. `catch` and `finally` aren't important because they aren't actually checked in a Promise test if `then` is `undefined`.
Check
```js
Promise.resolve({ get catch() { console.log("catch") }, get then() { console.log("then") }, get finally() { console.log("finally") } })
```

Reply:
I don't believe it's necessary. `catch` and `finally` aren't important because they aren't actually checked in a Promise test if `then` is `undefined`.
Check
```js
Promise.resolve({ get catch() { console.log("catch") }, get then() { console.log("then") }, get finally() { console.log("finally") } })
```

Reply:
Oh nvm then. Learning something new everyday 😃